fedimintd/src/ui.rs

```diff
@@ -301,7 +301,8 @@ async fn post_federation_params(
             name: form.guardian_name,
             tls_connect_string,
         },
-        finality_delay: form.finality_delay,
+        // finality delay is always one less than required block confirmations
+        finality_delay: form.block_confirmations - 1,
```

Comment:
This will panic if `block_confirmations` is 0. Front end validation should not be considered secure. Use [`checked_sub`](https://doc.rust-lang.org/std/primitive.u32.html#method.checked_sub) instead to generate an error return value instead of a panic.